### ISSUE TITLE
Precise editor cursor position descriptions

### ISSUE THREAD

**Issue by amyjko** [wordplaydev/wordplay#311]
## Expected behavior

When using VoiceOver in the code editor, the screen reader should announce where exactly in a token the caret is.

## Actual behavior

1. Create a project with the code `"first" + "second"`
2. Place the cursor after `sec`
3. The screen reader says "inside second"

It should say something lik "inside second, between c and o" or some other precise description to allow for editing.

## Environment

- mac OS 14.0
- Chrome 119
- VoiceOver activated
